src/highlight/__tests__/HighlightCreatorManager-test.ts

```diff
@@ -0,0 +1,135 @@
+    };
+    const rootEl = document.createElement('div');
+    const getOptions = (options: Partial<Options> = {}): Options => ({
+        ...defaults,
+        referenceEl: rootEl.querySelector('.reference') as HTMLElement,
+        ...options,
+    });
+    const getLayer = (): HTMLElement => rootEl.querySelector('.reference') as HTMLElement;
+    const getWrapper = (options?: Partial<Options>): HighlightCreatorManager =>
+        new HighlightCreatorManager(getOptions(options));
+
+    const mockReferenceEl = document.createElement('div');
+
+    beforeEach(() => {
+        rootEl.classList.add('root');
+        rootEl.innerHTML = '<div class="reference" />'; // referenceEl
+    });
+
+    describe('constructor()', () => {
+        test('should set all necessary properties', () => {
+            const wrapper = getWrapper();
+
+            expect(wrapper.referenceEl).toEqual(getLayer());
+        });
+    });
+
+    describe('destroy()', () => {
+        test('should clear timeout and remove event handlers', () => {
+            const wrapper = getWrapper();
+
+            wrapper.referenceEl = mockReferenceEl;
+            wrapper.selectionChangeTimer = 1;
+
+            jest.spyOn(mockReferenceEl, 'removeEventListener');
+
+            wrapper.destroy();
+
+            expect(clearTimeout).toHaveBeenCalledWith(1);
+            expect(mockReferenceEl.removeEventListener).toHaveBeenCalledWith('mousedown', wrapper.handleMouseDown);
+            expect(mockReferenceEl.removeEventListener).toHaveBeenCalledWith('mouseup', wrapper.handleMouseUp);
+        });
+    });
+
+    describe('exists()', () => {
+        test('should return a boolean based on its presence in the page element', () => {
+            const wrapper = getWrapper();
+
+            expect(wrapper.exists(rootEl)).toBe(true);
+            expect(wrapper.exists(document.createElement('div'))).toBe(false);
+        });
+    });
+
+    describe('render()', () => {
+        test('should clear previous selection and add listeners', () => {
+            const wrapper = getWrapper();
+
+            wrapper.referenceEl = mockReferenceEl;
+
+            jest.spyOn(mockReferenceEl, 'addEventListener');
+
+            wrapper.render();
+
+            expect(defaults.store.dispatch).toHaveBeenCalled();
+            expect(mockReferenceEl.addEventListener).toHaveBeenCalledWith('mousedown', wrapper.handleMouseDown);
+            expect(mockReferenceEl.addEventListener).toHaveBeenCalledWith('mouseup', wrapper.handleMouseUp);
+        });
+    });
+
+    describe('handleMouseDown()', () => {
+        test('should clear timeout and selection', () => {
+            const wrapper = getWrapper();
+
+            wrapper.selectionChangeTimer = 1;
+
+            wrapper.handleMouseDown(new MouseEvent('mousedown', { buttons: 1 }));
+
+            expect(setIsSelectingAction).toHaveBeenCalled();
+            expect(defaults.store.dispatch).toHaveBeenCalled();
```

Comment:
Is the order relevant? If so, should the dispatch assertions check to see what they were called with and in what order?